[WIP] Update Readme Content

tonydewan opened the pull request:
- [x] update content to better describe ISD
- [x] update the list of services
- [x] update the system metrics section
- [x] add links to website docs where appropriate

NOTE: this is blocked by merging of the documentation PR in `instrumental_web` because it contains links that will result in a 404 in production.

## TODO

~~How much duplication should we have between the website docs and various READMEs in the repo? I think both should be comprehensive, though this creates a potential maintenance issue later. To that end, we should probably add the following files to enumerate the metrics ISD collects:~~

- ~~system-metrics.md~~
- ~~service-metrics.md~~

janxious commented:
I am counter your position regarding having both website and repo readmes be comprehensive. It's a lot of work to make either complete, and they have different formatting. We punted on this on DocRaptor and it seems to work totally okay.

👍 to the content here.

tonydewan commented:
> We punted on this on DocRaptor and it seems to work totally okay.

Did we? Which is comprehensive?

janxious commented:
The DocRaptor website is comprehensive. At least when we reference it from any of the agents

tonydewan commented:
Is your opinion that the website should be the comprehensive source in this case?

janxious commented:
Most def. In repo would be nice, but we have a lot more control over style and presentation on our website, as well as better linkability to make things for people. Changing custom documentation presentation to fit into plain markdown is like, not fun.

tonydewan commented:
> The DocRaptor website is comprehensive. At least when we reference it from any of the agents

I'm not sure about this. The part of the DR website about the agents includes exactly the same content of the repos minus the development/release/version sections. Example: [PHP agent repo](https://github.com/DocRaptor/docraptor-php) and [website docs](https://docraptor.com/documentation/php). 

IIRC, the maintenance difficulty of content duplication is reduced by a rake task. That may or may not be necessary or reasonable here.

tonydewan commented:
>  Changing custom documentation presentation to fit into plain markdown is like, not fun.

I definitely agree with you on this point, and came into this with the opinion that the website should be the single source of truth. I changed my mind when I made this PR, but have now changed it back upon deeper consideration.

janxious commented:
I mean, I agree the agent details are similar but the details of how to interact with the agent in terms of options is all in the API documentation, which we do not reproduce, aside from a yaml file that I would not reasonably expect end users to peruse.

esquivalient commented:
This is everything I dreamed it could be.

mediocretes commented on README.md:
```diff
@@ -2,36 +2,50 @@
 
 ## Installation
-`instrumentald` is currently officially supported on 32-bit and 64-bit Linux, Windows systems and Mac OS X. There are prebuilt packages available for Debian, Ubuntu, RHEL and Win32 systems.
+`instrumentald` is supported on 32-bit and 64-bit Linux, Windows and OSX/macOS. There are prebuilt packages available for Debian, Ubuntu, RHEL and Win32 systems.
```
I think "/macOS" is redundant and confusing.  What is macOS, if not OSX?

tonydewan replied:
As of the soon to be released next version, it's called macOS. Old versions will likely still be called OSX. This is future proofing. If you want to reduce it, we should make it macOS only.

mediocretes replied:
TIL

tonydewan replied:
A link that is relevant: http://www.apple.com/macos/sierra-preview/

mediocretes replied:
I definitely remember knowing these facts at some point in the past, but I do not actually remember them.  Thanks for the link!

mediocretes commented:
:+1:

JamesPaden commented:
Merging with broken links that will be fixed when docs are updated.
